crates/biome_js_analyze/src/semantic_analyzers/nursery/no_exclusive_tests.rs

```diff
@@ -0,0 +1,149 @@
+use biome_analyze::{
+    context::RuleContext, declare_rule, Ast, FixKind, Rule, RuleDiagnostic, RuleSource,
+    RuleSourceKind,
+};
+use biome_console::markup;
+use biome_diagnostics::Applicability;
+use biome_js_syntax::{JsCallExpression, TextRange};
+use biome_rowan::{AstNode, BatchMutationExt, NodeOrToken};
+
+use crate::JsRuleAction;
+
+declare_rule! {
+    /// Disallow exclusive tests.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// describe.only("foo", function () {});
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// it.only("foo", function () {});
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// test.only("foo", function () {});
+    /// ```
+    ///
+    /// ### Valid
+    /// ```js
+    /// test("foo", function () {});
+    /// ```
+    ///
+    /// ```js
+    /// it("foo", function () {});
+    /// ```
+    ///
+    /// ```js
+    /// test("foo", function () {});
+    /// ```
+    pub NoExclusiveTests {
+        version: "next",
+        name: "noExclusiveTests",
+        recommended: true,
+        source: RuleSource::EslintJest("no-exclusive-tests"),
+        source_kind: RuleSourceKind::Inspired,
+        fix_kind: FixKind::Unsafe,
+    }
+}
+
+const FUNCTION_NAMES: [&str; 1] = ["only"];
+const CALEE_NAMES: [&str; 3] = ["describe", "it", "test"];
+
+impl Rule for NoExclusiveTests {
+    type Query = Ast<JsCallExpression>;
```

Comment:
I think this rule should be in `crates/biome_js_analyze/src/analyzers/nursery` since you use the Ast, not `Semantic`.
https://github.com/biomejs/biome/blob/main/crates/biome_analyze/CONTRIBUTING.md?rgh-link-date=2023-08-23T12%3A52%3A31Z#:~:text=type%20Query%20%3D%20Ast%3C%3E%20%2D%3E%20analyzers/%20folder